integration-tests/src/tests/network/runner.rs

```diff
@@ -27,20 +28,50 @@ use near_network::types::PeerManagerMessageRequest;
 use near_network::types::{NetworkRequests, NetworkResponses};
 use near_network::PeerManagerActor;
 use near_network_primitives::types::{
-    NetworkConfig, OutboundTcpConnect, PeerInfo, ROUTED_MESSAGE_TTL,
+    NetworkConfig, OutboundTcpConnect, PeerInfo, Ping as NetPing, Pong as NetPong,
+    ROUTED_MESSAGE_TTL,
 };
 use near_primitives::network::PeerId;
 use near_primitives::types::{AccountId, ValidatorId};
 use near_primitives::validator_signer::InMemoryValidatorSigner;
 use near_store::test_utils::create_test_store;
 use near_telemetry::{TelemetryActor, TelemetryConfig};
+use std::hash::Hash;
 use std::pin::Pin;
 
 pub type ControlFlow = std::ops::ControlFlow<()>;
 pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + 'a>>;
 pub type ActionFn =
     Box<dyn for<'a> Fn(&'a mut RunningInfo) -> BoxFuture<'a, anyhow::Result<ControlFlow>>>;
 
+#[derive(Debug, PartialEq, Eq)]
+struct MultiSet<T: Hash + Eq>(HashMap<T, usize>);
```

Comment:
My instinct is to always put such abstract utilities into separate module, to make sure then don't become entangled with specifics over time. No strong opinion though

Reply:
not if you want to derive PartialEq

Reply:
Oh wow, that's a cute hack, til, thanks!

Reply:
Yeah, the goal here is not to make the thing eventually reusable, but rather to make it clear to the reader (and potential future refactorer) that the API surface is indeed minuscule and unrelated to the surrounding code. The benefit here is that the reader immediately  understands "oh, that's a generic multiset, don't have to read that" and they don't need to read the impl to realise that there isn't anything specific to network there. 

From a different angle, often the "split or keep in a single file" question boils down to  size, and that seems wrong to me. It feels better to keep independent things separate, and dependent things together, mostly regardless of the size, and abstract utilities are the clearest example of independent stuff. 

No strong opinion here though!

Reply:
you are right, I'll put MultiSet into a separate file.